### PR TITLE
Home: Adjust location lock position for snackbar.

### PR DESCRIPTION
**Pull request by scolsen**
Updates our snackbar popup to support two enhancements:

1. The location lock button moves dynamically so that it is not obscured by the snackbar
2. The snackbar has a dismiss action which, on tap, dismisses the snackbar.

Towards #2285

Screenshot 2024-03-06 at 12 49 52 PM

### CODE REVIEW

**Comment by scolsen** [comment]
@gino-m I realize the mocks have an "X" button instead of "Dismiss" text. Action text is supported out of the box for snackbars, but to use a drawable we'll prob have to create our own custom snackbar view, which I don't have time for this week. don't know if you want to move ahead with this as a first impl and switch to the button later, or do something else

**Comment by gino-m** [comment]
> @gino-m I realize the mocks have an "X" button instead of "Dismiss" text. Action text is supported out of the box for snackbars, but to use a drawable we'll prob have to create our own custom snackbar view, which I don't have time for this week. don't know if you want to move ahead with this as a first impl and switch to the button later, or do something else

This is great, thanks Scott!

File: ground/src/main/java/com/google/android/ground/ui/common/EphemeralPopups.kt [hunk 1]
```diff
@@ -61,24 +61,29 @@ class EphemeralPopups @Inject constructor(private val context: Application) {
...
-      @StringRes messageId: Int,
-      duration: PopupDuration = PopupDuration.INDEFINITE,
-    ) {
-      val msg = context.resources.getString(messageId)
-      showSnackbar(view, msg, duration)
-    }
+  inner class InfoPopup(view: View, @StringRes messageId: Int, duration: PopupDuration) {
+    private var snackbar: Snackbar =
+      Snackbar.make(view, context.resources.getString(messageId), durationToSnackDuration(duration))
+        .setAction("Dismiss") { this.dismiss() }
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/common/EphemeralPopups.kt:67]: Please move to strings.xml

**Review by gino-m — CHANGES_REQUESTED** [review]

**Comment by scolsen** [comment]
@gino-m done and done!

**Review by gino-m — APPROVED** [review]
Nice!

**Pull request merged**